pypeit/fluxspec.py

```diff
@@ -66,6 +66,9 @@ class FluxSpec():
     # Frametype is a class attribute
     frametype = 'sensfunc'
 
+    # JFH TODO In my opinion the things that the class operates on should be arguments and the parameters
+    # should be in the parset. I really don't like to see parsets used in this way where everything is passed in
+    # under the hood. It makes it challenging to understand what the code is doing.
```

Comment:
Well the issue here is how the fluxing code works. Everything gets ingested by a pypeit file which is why we just pack all arguments to the parset in this way. I'm not sure I agree with that approach, but I don't know of a better way to pass arguments to the script, since the number of arguments is sufificnetly larger that using the parser would be cumbersome. 

Maybe we should have parsets that unpack information from a pypeit file and the pass them as arguments to the class. The class should not just have a parset as a global argument container, since if you try to run from i.e. a notebook as you often claim you do, you will have problems with this format.

Reply:
I don't follow, but am deferring.